src/probnum/problems/zoo/filtsmooth/_filtsmooth_problems.py

```diff
@@ -450,15 +517,23 @@ def l(t):
         measmod=measurement_model,
         initrv=initrv,
         times=time_grid,
+        random_state=random_state,
+    )
+    regression_problem = problems.TimeSeriesRegressionProblem(
+        observations=obs,
+        locations=time_grid,
+        measurement_models=measurement_model,
+        solution=states,
     )
-    regression_problem = problems.RegressionProblem(
-        observations=obs, locations=time_grid, solution=states
+    prior_process = randprocs.MarkovProcess(
+        transition=dynamics_model, initrv=initrv, initarg=time_grid[0]
     )
 
     statespace_components = dict(
         dynamics_model=dynamics_model,
         measurement_model=measurement_model,
```

Comment:
We don't have to return the measurement model anymore, right?

Reply:
Since it's part of the `RegressionProblem`

Reply:
we don't even have to return initrv and dynamics model. Shall I remove all of these, too?

Reply:
👍